app/models/manageiq/providers/openstack/inventory/collector/target_collection.rb

```diff
@@ -0,0 +1,212 @@
+class ManageIQ::Providers::Openstack::Inventory::Collector::TargetCollection < ManageIQ::Providers::Openstack::Inventory::Collector
+  def initialize(_manager, _target)
+    super
+    parse_targets!
+    infer_related_ems_refs!
+
+    # Reset the target cache, so we can access new targets inside
+    target.manager_refs_by_association_reset
+  end
+
+  def references(collection)
+    target.manager_refs_by_association.try(:[], collection).try(:[], :ems_ref).try(:to_a) || []
+  end
+
+  def name_references(collection)
+    target.manager_refs_by_association.try(:[], collection).try(:[], :name).try(:to_a) || []
+  end
+
+  def availability_zones
+    return [] if references(:availability_zones).blank?
+    @availability_zones = references(:availability_zones).collect do |az|
+      OpenStruct.new(:zoneName => az)
+    end
+  end
+
+  def cloud_networks
+    return [] if references(:cloud_networks).blank?
+    @cloud_networks = references(:cloud_networks).collect do |network_id|
+      safe_get { network_service.networks.get(network_id) }
```

Comment:
hm, could we use a filtered handled list? OpenStack should support list filtering by UUIDs right? In larger envs and e.g. bigger stack deployment, we can have hundreds of entities here. So that would a lot of API calls